database-plugins/src/main/java/co/cask/hydrator/plugin/DBRecord.java

```diff
@@ -207,12 +224,11 @@ private void writeToDB(PreparedStatement stmt, Schema.Type fieldType, @Nullable
         stmt.setBoolean(sqlIndex, (Boolean) fieldValue);
         break;
       case INT:
-        // write short or int appropriately
-        writeInt(stmt, fieldIndex, sqlIndex, fieldValue);
+        writeIntOrDate(stmt, fieldName, fieldValue, fieldIndex, sqlIndex);
```

Comment:
methods like this are a bad pattern. They indicate that the condition being checked is happening in the wrong place.

instead of introducing writeIntOrDate, refactor this method to take a Schema.Field instead of a Schema.Type and field name. Then check for logical type before checking physical type. Once we get here, we should always be writing an int.

Reply:
Thanks for the suggestion!